File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/IssuerSerialNumber.cs
using System;
using System.Runtime.Serialization;

//	IssuerSerialNumber ::= SEQUENCE {
//		issuer			Name,
//		serialNumber	INTEGER 
//	}

namespace Aladdin.ASN1.ISO.PKIX
{
	[Serializable]
	public class IssuerSerialNumber : Sequence
	{
		// информация о структуре
		private static readonly ObjectInfo[] info = new ObjectInfo[] { 

			new ObjectInfo(new ChoiceCreator<Name	>().Factory(), Cast.N), 
			new ObjectInfo(new ObjectCreator<Integer>().Factory(), Cast.N), 
		}; 
		// конструктор при сериализации
        protected IssuerSerialNumber(SerializationInfo info, StreamingContext context) : base(info, context) {}

		// конструктор при раскодировании
		public IssuerSerialNumber(IEncodable encodable) : base(encodable, info) {}

		// конструктор при закодировании
		public IssuerSerialNumber(IEncodable issuer, Integer serialNumber) : 
			base(info, issuer, serialNumber) {}

		public IEncodable	Issuer			{ get { return			 this[0]; } } 
		public Integer		SerialNumber	{ get { return (Integer	)this[1]; } }
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/IssuerSerialNumber.java
package aladdin.asn1.iso.pkix;
import aladdin.asn1.*; 
import aladdin.asn1.Integer; 
import java.io.*; 

//	IssuerSerialNumber ::= SEQUENCE {
//		issuer			Name,
//		serialNumber	INTEGER 
//	}

public final class IssuerSerialNumber extends Sequence<IEncodable>
{
    private static final long serialVersionUID = -765807028901536567L;
    
	// информация о структуре
	private static final ObjectInfo[] info = new ObjectInfo[] { 

		new ObjectInfo(new ChoiceCreator(Name     .class).factory(), Cast.N), 
		new ObjectInfo(new ObjectCreator(Integer  .class).factory(), Cast.N), 
	}; 
	// конструктор при раскодировании
	public IssuerSerialNumber(IEncodable encodable) throws IOException { super(encodable, info); }

	// конструктор при закодировании
	public IssuerSerialNumber(IEncodable issuer, Integer serialNumber) 
	{
		super(info, issuer, serialNumber); 
	}
	public final IEncodable	issuer	    () { return		     get(0); } 
	public final Integer    serialNumber() { return (Integer)get(1); }
}
